### ISSUE TITLE
Feature requests: Cloud Storage and Triggered bonuses

### ISSUE THREAD

**Issue by FatherPrax** [zombieFox/awesomeSheet#174]
Great tool, love it. 2 areas I'm having issues that I think could be worthwhile, and I didn't see requested before. Also, is the plan to keep the public facing version of the website up as long as you keep the project going for people to use? OR are you expecting everyone who uses it to build their own version of the website and host it? I've done that for myself, and will have others in my group use my hosted version, but just wondering.

1) Import/Export JSON file. This works, and I will continue to use this to keep up with the character on multiple systems (tablet during play, PC when leveling up/etc between sessions). However, if it could have an option for integration with a cloud storage provider to read/write the character sheet to, it would be much easier. Tying instead to a json file stored on google drive or other provider. Nitpicky, but would be a useful thing, and that is how these kinds of things are done nowadays in a ton of business apps.

2) Conditional Bonuses/Drawbacks. Have a Ranger I'm currently playing who has like 5 different types of conditions that affect both skills and attacks. I currently just have 5 lines of conditions on my skills tab, but looking thru every time I roll to make sure "Okay, IS a favored enemy, NOT a favored terrain, IS tracking (vs regular survival), that means I have a +2 on these skills, +3 on these, a +1 on those 2..." etc is a pain. A section where you can click in customizeable conditions would be hugely beneficial. Also works for negatives or non class specific stuff like "Frightened: -2 on Attack, Skills and abilities" and have it autocalculate while you have that condition toggled.

**Comment by zombieFox** [comment]
Hi, glad you are finding awesomeSheet useful. 👍 

The plan is indeed to keep the public facing application live for as long as github allows free hosting. A user of awesomeSheet should **not** have to run the project locally. awesomeSheet is a progressive web app and should run fine with [github hosting it](https://zombiefox.github.io/awesomeSheet/). Nothing wrong with hosting it yourself, mind you. It's just more work.

1. This is not the first time this feature has been requested. I've posted the project a few times on [/r/Pathfinder_RPG/](https://www.reddit.com/r/Pathfinder_RPG/) and it's something that keeps being pointed out. The issue is I'm not that kind of developer and would need a lot of support to get a feature like that off the ground. I've even spoken to some developer friends and am investigating how this could be done with minimal work -- at least with google accounts to start with.

Also, If you check out my posts on reddit you will see that major feature development is on hold. This is because of major (good) life events at the moment. As such I can't commit nearly as much time as I used to on awesomeSheet.

2. This has also been requested a few times. Please see issue #168 as it's the closest to this. I agree, more automation would be helpful, I've hit this same issue using awesomeSheet too. Unfortunately this falls under the "major feature request" category so will be on hold for a while.

I appreciate that all the answers above amount to "sounds cool, but can't do it yet". So sorry for that. Hopefully in the future I will be able to pick up these ideas and run with them.

**Comment by Xuroth** [comment]
I apologize if this is the wrong place to inquire (I'm sure it is). I am developing my own character manager (that will feature cloud storage). I am VERY early in development (I've spun up a few simple test version, none were feature complete (or indeed, functional)) but I have all the tools and resources now to begin work on it (as a hobby). The repository is private, and I plan on charging some users for advanced tools and special features (just mentioning this in case it's an issue).

One of the FREE features I want to implement is Import to/Export from AwesomeSheet, so if they create/clone a character and want to use it in your (awesome) sheet, they can. Or vice versa, if they want to put their AS character into my character manager to get access to cloud storage and such. 

Sorry for the long post, but there IS a question:

TLDR - Would I have permission to use your logo for the Import/Export functionality?

**Comment by zombieFox** [comment]
@Xuroth Hi, that sounds like an exciting project. Sure, I don't mind. Feel free to use the [PSD](https://github.com/zombieFox/awesomeSheet/tree/master/src/psd) found in the repo.

I'd be interested in the project, would there be anyway to follow it?

**Comment by Xuroth** [comment]
Thanks for your response. I'll use the PSD file. I'd be delighted to invite you to the private repo once I get a bit more work done on the main functionality. Currently I'm working on user roles and authentication.
